primitives/src/condition.rs

```diff
@@ -139,8 +151,36 @@ impl Condition {
     }
 
     /// Returns worst case complexity of a condition.
-    pub fn complexity(&self) -> (usize, usize) {
-        (self.condition_type.complexity(), self.issuers.len())
+    pub fn complexity(&self, default_issuer_count: usize) -> u32 {
+        let issuers = match self.issuers.len() {
+            0 => default_issuer_count,
+            count => count,
+        };
+        self.condition_type
+            .count()
+            // NB: `max(1)` Make sure issuer count is not zero.
```

Comment:
Also, this could use some more explaining as to the *why* we want to make sure it's not zero (beyond the mathematical fact that the multiplication => 0).

Reply:
It is fine as is.